### PR TITLE
feat(components/pages): add blocks, list, and tabs page layouts

### PR DESCRIPTION
**Pull request by Blackbaud-PaulCrowder**
[AB#2537855](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2537855)
[AB#2537858](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2537858)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1377/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/forms-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/1377/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1377/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1377/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1377/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1377/playground/)

File: libs/components/pages/src/lib/modules/page/types/page-layout-type.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export type SkyPageLayoutType =
+  | 'auto'
+  | 'fit'
```
Comment (Blackbaud-SteveBrush) [inline libs/components/pages/src/lib/modules/page/types/page-layout-type.ts]: Just for clarity, we would never have a page that "fits" its layout, but uses a "box" layout?

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
Do we want to add code examples in a later PR?

File: libs/components/pages/src/lib/modules/page/types/page-layout-type.ts (hunk 1 above)
Comment (Blackbaud-PaulCrowder) [reply libs/components/pages/src/lib/modules/page/types/page-layout-type.ts]: `auto` and `fit` are general-purpose layout types, whereas `box`, `list`, `split-view` and `tabs` are specifically to support a handful of page layouts using SKY UX components as their main content. I added some code comments to clarify that a bit.

**Comment by Blackbaud-MattGregg** [comment]
@Blackbaud-PaulCrowder The use of new sky-page-header class on the H1 page header - is this going to one of the things that consumers will have to remember?
We hadn't included a header container after we stopped trying to figure the variants for header content and just focused on page-level outer spacing. This is including a sky-page-header-content with set spacing. Why is this needed vs just having a page header container for everything that could be in there?

**Comment by Blackbaud-PaulCrowder** [comment]
> @Blackbaud-PaulCrowder The use of new sky-page-header class on the H1 page header - is this going to one of the things that consumers will have to remember? We hadn't included a header container after we stopped trying to figure the variants for header content and just focused on page-level outer spacing. This is including a sky-page-header-content with set spacing. Why is this needed vs just having a page header container for everything that could be in there?

The element has that CSS class, but it's put there by the `sky-page-header` component. The consumer only specifies a string for the heading text on the component and we generate the `h1` element.

**Comment by Blackbaud-MattGregg** [comment]
For split view (fit)
Not sure why I didn't catch this before but split view is 'fit' but uses the same margins as lists (right and left-side). It doesn't go all the way to the page edge. Check out the chart for <sky-page-content>. There aren't any page that fit completely to the left, right edges.

**Comment by Blackbaud-MattGregg** [comment]
For split view - that page design does have a secondary button under the header in the section you currently have 'Some content'. Button would be more realistic example if you want to have that in example. https://developer.blackbaud.com/skyux/design/guidelines/page-layouts/split-view-page

**Review by Blackbaud-ErikaMcVey — DISMISSED** [review]

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

**Pull request merged**